app/models/wallet_history.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class WalletHistory < ApplicationRecord
+  belongs_to :wallet
+  validates :recepient_name, presence: true
+  validates :points, format: { with: /\A-?\d+\Z/ }
+  validates :transaction_type, inclusion: { in: %w[Referral Unlocked Like] }
```

Comment:
Move `[Referral Unlocked Like]` to a Constant...
Because later on we will be using these types at multiple places.
And if in future type changes then we need to make the change in those many places.